src/utils/DataProcessingUtils.js

```diff
@@ -230,22 +230,14 @@ function processEntityValue(key :string, value :any, mappers :Map | Object) :any
   }
 
   if (isDefined(processedValue)) {
-    if (isValidDataPrimitive(processedValue)) {
-      return [processedValue];
-    }
-    if (isValidDataPrimitiveArray(processedValue)) {
+    if (Array.isArray(processedValue) || List.isList(processedValue)) {
```

Comment:
I feel like this is a bug. If `processedValue` is indeed an immutable `List`, then this will return it as is. I don't see any `.toJS()` in the flow, so wouldn't this mean the end result would contain `List`?

Reply:
updated.

Reply:
I suspect we will eventually remove immutable as a dependency entirely. It's only in the repo as an artifact from the milestone project.